app/controllers/users_controller.rb

```diff
@@ -4,9 +4,15 @@ def new
   end
 
   def create
-    new_user = User.create(user_params)
-    flash[:success] = "Hello #{new_user.first_name} #{new_user.last_name}, thank you for creating an account!"
-    redirect_to "/profile"
+    new_user = User.new(user_params)
+    if new_user.save
+      flash[:success] = "Hello #{new_user.first_name} #{new_user.last_name}, thank you for creating an account!"
+      session[:user_id] = new_user.id
+      redirect_to "/profile"
+    else
+      flash[:notice] = new_user.errors.full_messages.join(". ").to_s
```

Comment:
Is there a better way to do this?

Reply:
Jack used `flash[:error] = @obituary.errors.full_messages.to_sentence` in the ObituariesController and I think that accomplishes the same thing? @jpc20 please correct me if I'm wrong!

Reply:
It is my understanding that you can name this anything, ie flash[:troyandgabriella].

Reply:
but error and notice are conventional, so stick with that.